Source/options.js

```diff
@@ -1,27 +1,129 @@
+    var fuelValue = fuel.value;
+    var distance = document.getElementById('distanceInput1');
+    var distanceValue = distance.value;
+    var input = document.getElementById('fuelConsumptionInput');
+    var inputValue = input.value;
+    var status = document.getElementById('save-message');
+    status.innerHTML = 'Saved!';
+    switch (fuelValue) {
+        case "l":
+            if (distance == "km") {
+                console.log("l/km case");
+                var fuelConsumption = 2328 * inputValue;
+                localStorage['carbonEmission'] = fuelConsumption;
+            } else {
+                var fuelConsumption = 2328 * inputValue;
+                localStorage['carbonEmission'] = fuelConsumption;
+            }
+            break;
+        case "g":
+            if (distance == "km") {
+                var fuelConsumption = 8782.155 * inputValue;
+                localStorage['carbonEmission'] = fuelConsumption;
+            } else {
+                var fuelConsumption = 1 / inputValue;
+                fuelConsumption = 5456.97 * fuelConsumption;
+            }
 
-function saveOptions() {
-  var emission = document.getElementById('emission');
-  var carbonEmission = emission.value;
-  localStorage['carbonEmission'] = carbonEmission;
-
-  // Update status to let user know options were saved.
-  var status = document.getElementById('save-message');
-  status.innerHTML = 'Saved!';
-  setTimeout(function() {
-    status.innerHTML = '';
-  }, 750);
-}
+            break;
+    }
+});
 
-function S(key) { return localStorage[key]; }
+$('#save-button-2').click(function() {
+    var unit = document.getElementById('distanceFuelInput2');
+    var unitValue = unit.value;
+    var input = document.getElementById('fuelEfficiencyInput');
+    var inputValue = input.value;
+    var status = document.getElementById('save-message-2');
+    status.innerHTML = 'Saved!';
+    if (unitValue == "km/l") {
+        var fuelEfficiency = 1 / inputValue;
+        fuelEfficiency = 2328 * fuelEfficiency;
+        localStorage['carbonEmission'] = fuelEfficiency;
+    } else if (unitValue == "km/gl") {
+        var fuelEfficiency = 1 / inputValue;
+        fuelEfficiency = 8782.155 * fuelEfficiency;
+        localStorage['carbonEmission'] = fuelEfficiency;
+    } else {
+        var fuelEfficiency = 5456.97 * inputValue;
+        localStorage['carbonEmission'] = fuelEfficiency;
+    }
+});
 
-function restoreOptions() {
-  var emission = document.getElementById('emission');
-  emission.setAttribute('value', S('carbonEmission'));
-}
+$('#save-button-3').click(function() {
+
+    var fuel = document.getElementById('fuelInput3');
+    var fuelValue = fuel.value;
+    var distance = document.getElementById('distanceInput3');
+    var distanceValue = distance.value;
+    var input = document.getElementById('co2EmissionInput');
+    var inputValue = input.value;
+    var status = document.getElementById('save-message-3');
```

Comment:
The previous id is in camel case and this one has dashes, please use a consistent convention. Preferably use dashes (just a personal preference)